api/src/main/java/com/graphhopper/GHRequest.java

```diff
@@ -41,7 +44,7 @@
     private List<String> pointHints = new ArrayList<>();
     private List<String> curbsides = new ArrayList<>();
     private List<String> snapPreventions = new ArrayList<>();
-    private List<String> pathDetails = new ArrayList<>();
```

Comment:
I'm unsure if we should make it a set, especially since the request order (see below) might be significant.

Same problem in Request.pathDetails.